packages/financial-templates-lib/test/clients/ExpiringMultiPartyClient.js

```diff
@@ -3,17 +3,20 @@ const { parseFixed } = require("@ethersproject/bignumber");
 const winston = require("winston");
 
 const { interfaceName, MAX_UINT_VAL, ZERO_ADDRESS } = require("@uma/common");
+const { getTruffleContract } = require("@uma/core");
+
+const ABI_VERSION = "1.2.0";
 
 const { ExpiringMultiPartyClient } = require("../../src/clients/ExpiringMultiPartyClient");
 
-const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
-const Finder = artifacts.require("Finder");
-const IdentifierWhitelist = artifacts.require("IdentifierWhitelist");
-const MockOracle = artifacts.require("MockOracle");
-const Token = artifacts.require("ExpandedERC20");
-const SyntheticToken = artifacts.require("SyntheticToken");
-const Timer = artifacts.require("Timer");
-const Store = artifacts.require("Store");
+const ExpiringMultiParty = getTruffleContract("ExpiringMultiParty", web3, ABI_VERSION);
```

Comment:
Is the idea to eventually ripple this change through all the tests? I'm curious how we see this scaling. For example, if we make ABI or logic breaking changes between upgrades (as we've done with the most recent update to the EMP with withdraw liquidation logic) we would need to add some kind of branching logic within the tests to accommodate different sections based on the ABI version. This is with a _small_ change to the contracts. what happens when there is a larger difference between versions? 

I think this solution _might_ be ok for now but I'm hesitant to say this is the path we should go down for multi-version testing before we have a better strategy & plan for supporting these kinds of tests within the repo. As an example, It might be better to have separate test files for each version to make it clear what is being tested and to remove the need for any branching logic within the tests.

Reply:
I don't think I'm making a statement about how we should handle versioning within the EMPClient, in fact I'm hereby assuming that the EMPClient should only work with the 1.2 version. I think a subsequent change to the EMPClient should be made to change logic based on the versioning, but in a scalable smart way.

What do you think about placing ABI/CONTRACT_VERSION in the `ExpiringMultiPartyClient.js` file itself, exporting it as a variable, and importing that variable in the test? This way it is more clear what version the client is supposed to be working with?

Reply:
ye, that could work! There will be situations wherein the client can support multiple versions; say for example we release 1.3.0 with the new contracts (when they are on main net). We could then define the EMPClient to support `1.2.x` & `1.3.x`, with some small internal branching logic to deal with both cases. In this scenario, you could pull the "supported" versions from the client within the test and then check that they overlap with those used within the unit tests. we might then want to have a `1.2.x` section of tests and a separate section that tests `1.3.x`.

What do you think? I'm also curious for input on this point from @UMAprotocol/eng as this is the first example of where we need to consider versioning into the future and how we want to interplay within tests.

Reply:
I generally agree with @nicholaspai that this PR is just tweaking how the versions are accessed and making it more explicit. It's unopinionated wrt how multiple versions would be handled in the client/tests.

Reply:
@chrismaree i think the structure of the versioning tests depends on how different the versions are. For very similar versions, you could imagine wanting to run the exact same tests just with the version swapped. For more branch-y behavior, you might want one or two unique tests or even a separate test suite. Thoughts?

Reply:
> We could then define the EMPClient to support 1.2.x & 1.3.x, with some small internal branching logic to deal with both cases.

I think we should build internal branching logic yes.

I also think eventually we'll want to export, from all clients, the version strings that they are compatible with (e.g. EMPClient compatible with <= 1.2.2 and OOClient compatible with >= 1.2.5), and their corresponding unit tests should pass for all eligible verrsions (tbd how unit tests for different versions will work).

Additionally, we'll want to enforce within the downstream packages, like `liquidator` and `disputer`, that they are importing the correct versions to work with the clients. For example, we don't want the liquidator to use `@uma/core: 1.2.7` with the EMPClient if the EMPClient is only compatible with <= 1.2.5. TBD how we enforce this and whether we can do it dynamically.

That being said, these are all tangential to this PR and IMO are not features that should block it. This PR is mostly just bumping the `financial-templates-lib` package's @uma/core version to latest so that I can put an OptimisticOracleClient in it.

Reply:
That all sounds good & you are right this is unrelated to this PR. Did not mean to block it from going through! LGTM!

Reply:
Hey @chrismaree @mrice32 can you check out this [comment](https://github.com/UMAprotocol/protocol/pull/2344#discussion_r550182007) related to this thread.